Empty log file string crashes the server

Passing an empty log file string (`"xml.logs.file": ""`) via `initalizationOptions` crashes the language server.

The expected behavior is an empty string to be handled the same way as `null`, which just doesn't create a log file.

Traceback:

```
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
		at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:67)
		at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.request(GenericEndpoint.java:120)
		at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:261)
		at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:190)
		at org.eclipse.lemminx.commons.ParentProcessWatcher.lambda$apply$0(ParentProcessWatcher.java:143)
		at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
		at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
		at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
		at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
		at java.util.concurrent.FutureTask.run(Unknown Source)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
		at java.lang.Thread.run(Unknown Source)
	Caused by: java.lang.reflect.InvocationTargetException
		at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
		at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
		at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
		at java.lang.reflect.Method.invoke(Unknown Source)
		at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
		... 12 more
	Caused by: java.lang.IllegalArgumentException: Incorrect file path provided
		at org.eclipse.lemminx.logs.LogHelper.getFileHandler(LogHelper.java:114)
		at org.eclipse.lemminx.logs.LogHelper.initializeRootLogger(LogHelper.java:59)
		at org.eclipse.lemminx.XMLLanguageServer.updateSettings(XMLLanguageServer.java:150)
		at org.eclipse.lemminx.XMLLanguageServer.initialize(XMLLanguageServer.java:111)
		... 17 more
```

- LemMinX 0.13.1

--- a/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/logs/LogHelper.java
+++ b/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/logs/LogHelper.java
@@ -24,6 +24,7 @@ import java.util.logging.Logger;
 import java.util.logging.SimpleFormatter;
 
 import org.eclipse.lemminx.settings.LogsSettings;
+import org.eclipse.lemminx.utils.StringUtils;
 import org.eclipse.lsp4j.services.LanguageClient;
 
 /**
@@ -43,7 +44,7 @@ public class LogHelper {
 		unregisterAllHandlers(logger.getHandlers());
 		logger.setLevel(getLogLevel());
 		logger.setUseParentHandlers(false);// Stops output to console
-		
+
 		// Configure logging LSP client handler
 		if (settings != null) {
 			if (settings.isClient()) {
@@ -55,7 +56,7 @@ public class LogHelper {
 			}
 			// Configure logging for file
 			String path = settings.getFile();
-			if (path != null) {
+			if (!StringUtils.isBlank(path)) {
 				createDirectoryPath(path);
 				try {
 					FileHandler fh = LogHelper.getFileHandler(path);
@@ -68,7 +69,7 @@ public class LogHelper {
 			}
 		}
 	}
-	
+
 	private static Level getLogLevel() {
 		String logLevel = System.getProperty("log.level", "info").toLowerCase();
 		switch (logLevel) {

--- a/org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/MockXMLLanguageClient.java
+++ b/org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/MockXMLLanguageClient.java
@@ -1,0 +1,113 @@
+/*******************************************************************************
+* Copyright (c) 2020 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lemminx;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import org.eclipse.lemminx.customservice.ActionableNotification;
+import org.eclipse.lemminx.customservice.XMLLanguageClientAPI;
+import org.eclipse.lsp4j.ExecuteCommandParams;
+import org.eclipse.lsp4j.MessageActionItem;
+import org.eclipse.lsp4j.MessageParams;
+import org.eclipse.lsp4j.PublishDiagnosticsParams;
+import org.eclipse.lsp4j.RegistrationParams;
+import org.eclipse.lsp4j.ShowMessageRequestParams;
+import org.eclipse.lsp4j.UnregistrationParams;
+
+/**
+ * Mock XML Language client which helps to track show messages, actionable
+ * notification and commands.
+ * 
+ * @author Angelo ZERR
+ *
+ */
+public class MockXMLLanguageClient implements XMLLanguageClientAPI {
+
+	private final List<PublishDiagnosticsParams> publishDiagnostics;
+
+	private final List<MessageParams> showMessages;
+
+	private final List<ActionableNotification> actionableNotifications;
+
+	private final List<MessageParams> logMessages;
+
+	public MockXMLLanguageClient() {
+		publishDiagnostics = new ArrayList<>();
+		showMessages = new ArrayList<>();
+		logMessages = new ArrayList<>();
+		actionableNotifications = new ArrayList<>();
+	}
+
+	@Override
+	public CompletableFuture<MessageActionItem> showMessageRequest(ShowMessageRequestParams requestParams) {
+		return null;
+	}
+
+	@Override
+	public void showMessage(MessageParams messageParams) {
+		showMessages.add(messageParams);
+	}
+
+	@Override
+	public void publishDiagnostics(PublishDiagnosticsParams diagnostics) {
+		publishDiagnostics.add(diagnostics);
+	}
+
+	@Override
+	public void logMessage(MessageParams message) {
+		logMessages.add(message);
+	}
+
+	@Override
+	public void telemetryEvent(Object object) {
+
+	}
+
+	@Override
+	public void actionableNotification(ActionableNotification notification) {
+		actionableNotifications.add(notification);
+	}
+
+	@Override
+	public CompletableFuture<Void> registerCapability(RegistrationParams params) {
+		return CompletableFuture.completedFuture(null);
+	}
+
+	@Override
+	public CompletableFuture<Void> unregisterCapability(UnregistrationParams params) {
+		return CompletableFuture.completedFuture(null);
+	}
+
+	@Override
+	public CompletableFuture<Object> executeClientCommand(ExecuteCommandParams params) {
+		throw new UnsupportedOperationException();
+	}
+
+	public List<PublishDiagnosticsParams> getPublishDiagnostics() {
+		return publishDiagnostics;
+	}
+
+	public List<MessageParams> getLogMessages() {
+		return logMessages;
+	}
+
+	public List<MessageParams> getShowMessages() {
+		return showMessages;
+	}
+
+	public List<ActionableNotification> getActionableNotifications() {
+		return actionableNotifications;
+	}
+
+}

--- a/org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/MockXMLLanguageServer.java
+++ b/org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/MockXMLLanguageServer.java
@@ -11,21 +11,15 @@
 *******************************************************************************/
 package org.eclipse.lemminx;
 
-import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 import java.util.concurrent.CompletableFuture;
 
 import org.eclipse.lemminx.customservice.ActionableNotification;
-import org.eclipse.lemminx.customservice.XMLLanguageClientAPI;
 import org.eclipse.lemminx.services.extensions.commands.IXMLCommandService.IDelegateCommandHandler;
 import org.eclipse.lsp4j.ExecuteCommandParams;
-import org.eclipse.lsp4j.MessageActionItem;
 import org.eclipse.lsp4j.MessageParams;
 import org.eclipse.lsp4j.PublishDiagnosticsParams;
-import org.eclipse.lsp4j.RegistrationParams;
-import org.eclipse.lsp4j.ShowMessageRequestParams;
-import org.eclipse.lsp4j.UnregistrationParams;
 
 /**
  * Mock XML Language server which helps to track show messages, actionable
@@ -36,76 +30,25 @@ import org.eclipse.lsp4j.UnregistrationParams;
  */
 public class MockXMLLanguageServer extends XMLLanguageServer {
 
-	private final List<PublishDiagnosticsParams> publishDiagnostics;
-
-	private final List<MessageParams> showMessages;
-
-	protected final List<ActionableNotification> actionableNotifications;
-
 	public MockXMLLanguageServer() {
-		publishDiagnostics = new ArrayList<>();
-		showMessages = new ArrayList<>();
-		actionableNotifications = new ArrayList<>();
-		XMLLanguageClientAPI client = new XMLLanguageClientAPI() {
-
-			@Override
-			public CompletableFuture<MessageActionItem> showMessageRequest(ShowMessageRequestParams requestParams) {
-				return null;
-			}
-
-			@Override
-			public void showMessage(MessageParams messageParams) {
-				showMessages.add(messageParams);
-			}
-
-			@Override
-			public void publishDiagnostics(PublishDiagnosticsParams diagnostics) {
-				publishDiagnostics.add(diagnostics);
-			}
-
-			@Override
-			public void logMessage(MessageParams message) {
-
-			}
-
-			@Override
-			public void telemetryEvent(Object object) {
-
-			}
-
-			@Override
-			public void actionableNotification(ActionableNotification notification) {
-				actionableNotifications.add(notification);
-			}
-
-			@Override
-			public CompletableFuture<Void> registerCapability(RegistrationParams params) {
-				return CompletableFuture.completedFuture(null);
-			}
-
-			@Override
-			public CompletableFuture<Void> unregisterCapability(UnregistrationParams params) {
-				return CompletableFuture.completedFuture(null);
-			}
-
-			@Override
-			public CompletableFuture<Object> executeClientCommand(ExecuteCommandParams params) {
-				throw new UnsupportedOperationException();
-			}
-		};
-		super.setClient(client);
+		super.setClient(new MockXMLLanguageClient());
 	}
 
 	public List<PublishDiagnosticsParams> getPublishDiagnostics() {
-		return publishDiagnostics;
+		return getLanguageClient().getPublishDiagnostics();
+	}
+
+	@Override
+	public MockXMLLanguageClient getLanguageClient() {
+		return (MockXMLLanguageClient) super.getLanguageClient();
 	}
 
 	public List<MessageParams> getShowMessages() {
-		return showMessages;
+		return getLanguageClient().getShowMessages();
 	}
 
 	public List<ActionableNotification> getActionableNotifications() {
-		return actionableNotifications;
+		return getLanguageClient().getActionableNotifications();
 	}
 
 	public void registerCommand(String commandId, IDelegateCommandHandler handler) {

--- a/org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/logs/LogHelperTest.java
+++ b/org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/logs/LogHelperTest.java
@@ -1,0 +1,47 @@
+/*******************************************************************************
+* Copyright (c) 2020 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lemminx.logs;
+
+import org.eclipse.lemminx.MockXMLLanguageClient;
+import org.eclipse.lemminx.settings.LogsSettings;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test for {@link LogHelper}.
+ * 
+ * @author Angelo ZERR
+ *
+ */
+public class LogHelperTest {
+
+	@Test
+	public void emptyPath() {
+		// Empty file path should not throw an error.
+
+		MockXMLLanguageClient languageClient = new MockXMLLanguageClient();
+		LogsSettings settings = new LogsSettings();
+		// Enable client
+		settings.setClient(true);
+
+		// Enable log file
+		settings.setFile(null);
+		LogHelper.initializeRootLogger(languageClient, settings);
+
+		// Enable log file
+		settings.setFile("");
+		LogHelper.initializeRootLogger(languageClient, settings);
+
+		// Enable log file
+		settings.setFile("    ");
+		LogHelper.initializeRootLogger(languageClient, settings);
+	}
+}

--- a/org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/settings/capabilities/XMLCapabilitiesTest.java
+++ b/org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/settings/capabilities/XMLCapabilitiesTest.java
@@ -21,8 +21,8 @@ import static org.eclipse.lemminx.settings.capabilities.ServerCapabilitiesConsta
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 import java.util.Set;
-import java.util.concurrent.CompletableFuture;
 
+import org.eclipse.lemminx.MockXMLLanguageClient;
 import org.eclipse.lemminx.XMLTextDocumentService;
 import org.eclipse.lsp4j.ClientCapabilities;
 import org.eclipse.lsp4j.CodeActionCapabilities;
@@ -34,14 +34,9 @@ import org.eclipse.lsp4j.DocumentSymbolCapabilities;
 import org.eclipse.lsp4j.FoldingRangeCapabilities;
 import org.eclipse.lsp4j.FormattingCapabilities;
 import org.eclipse.lsp4j.HoverCapabilities;
-import org.eclipse.lsp4j.MessageActionItem;
-import org.eclipse.lsp4j.MessageParams;
-import org.eclipse.lsp4j.PublishDiagnosticsParams;
 import org.eclipse.lsp4j.RangeFormattingCapabilities;
-import org.eclipse.lsp4j.RegistrationParams;
 import org.eclipse.lsp4j.RenameCapabilities;
 import org.eclipse.lsp4j.ServerCapabilities;
-import org.eclipse.lsp4j.ShowMessageRequestParams;
 import org.eclipse.lsp4j.TextDocumentClientCapabilities;
 import org.eclipse.lsp4j.WorkspaceClientCapabilities;
 import org.eclipse.lsp4j.services.LanguageClient;
@@ -53,7 +48,7 @@ import org.junit.jupiter.api.Test;
  */
 public class XMLCapabilitiesTest {
 
-	private LanguageClient languageClient = new LanguageClientMock();
+	private LanguageClient languageClient = new MockXMLLanguageClient();
 	private XMLCapabilityManager manager;
 	private ClientCapabilities clientCapabilities;
 	private TextDocumentClientCapabilities textDocument;
@@ -224,31 +219,4 @@ public class XMLCapabilitiesTest {
 		capabilityIDs = manager.getRegisteredCapabilities();
 	}
 
-	class LanguageClientMock implements LanguageClient {
-		@Override
-		public void telemetryEvent(Object object) {
-		}
-
-		@Override
-		public void publishDiagnostics(PublishDiagnosticsParams diagnostics) {
-		}
-
-		@Override
-		public void showMessage(MessageParams messageParams) {
-		}
-
-		@Override
-		public CompletableFuture<MessageActionItem> showMessageRequest(ShowMessageRequestParams requestParams) {
-			return null;
-		}
-
-		@Override
-		public void logMessage(MessageParams message) {
-		}
-
-		@Override
-		public CompletableFuture<Void> registerCapability(RegistrationParams params) {
-			return null;
-		}
-	}
 }

--- a/org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/utils/LoggerTest.java
+++ b/org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/utils/LoggerTest.java
@@ -14,6 +14,7 @@ package org.eclipse.lemminx.utils;
 
 import static java.lang.System.lineSeparator;
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.junit.jupiter.api.Assertions.fail;
 
@@ -24,21 +25,17 @@ import java.io.IOException;
 import java.io.PrintWriter;
 import java.util.Locale;
 import java.util.TimeZone;
-import java.util.concurrent.CompletableFuture;
 import java.util.logging.Handler;
 import java.util.logging.Level;
 import java.util.logging.LogRecord;
 import java.util.logging.Logger;
 
+import org.eclipse.lemminx.MockXMLLanguageClient;
 import org.eclipse.lemminx.logs.LSPClientLogHandler;
 import org.eclipse.lemminx.logs.LogHelper;
 import org.eclipse.lemminx.settings.LogsSettings;
-import org.eclipse.lsp4j.MessageActionItem;
 import org.eclipse.lsp4j.MessageParams;
 import org.eclipse.lsp4j.MessageType;
-import org.eclipse.lsp4j.PublishDiagnosticsParams;
-import org.eclipse.lsp4j.ShowMessageRequestParams;
-import org.eclipse.lsp4j.services.LanguageClient;
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
@@ -49,7 +46,7 @@ import org.junit.jupiter.api.Test;
 public class LoggerTest {
 
 	private String path = "target/logs/testLogFile.log";
-	private MockLanguageClient mockLanguageClient;
+	private MockXMLLanguageClient mockLanguageClient;
 	private File logFile;
 	private Logger LOGGER = Logger.getLogger(LoggerTest.class.getName());
 	private TimeZone originalTimeZone = TimeZone.getDefault();
@@ -58,7 +55,7 @@ public class LoggerTest {
 	public void startup() {
 		TimeZone.setDefault(TimeZone.getTimeZone(("UTC")));
 		deleteLogFile();
-		mockLanguageClient = createLanguageClient(MessageType.Error, "Log Message");
+		mockLanguageClient = new MockXMLLanguageClient(); // (MessageType.Error, "Log Message");
 		LogsSettings settings = new LogsSettings();
 		// Enable log file
 		settings.setFile(path);
@@ -96,7 +93,10 @@ public class LoggerTest {
 	@Test
 	public void testClientReceivesLog() {
 		LOGGER.severe("@@Log Message@@");
-		assertTrue(mockLanguageClient.wasLogReceived());
+		assertFalse(mockLanguageClient.getLogMessages().isEmpty());
+		MessageParams message = mockLanguageClient.getLogMessages().get(0);
+		assertEquals(MessageType.Error, message.getType());
+		assertTrue(message.getMessage().endsWith("@@Log Message@@"));
 	}
 
 	@Test
@@ -142,50 +142,7 @@ public class LoggerTest {
 
 	// ---------------------Tools-------------------------------
 
-	private MockLanguageClient createLanguageClient(MessageType messageType, String message) {
-		MockLanguageClient newLanguageClient = new MockLanguageClient(messageType, message);
-		return newLanguageClient;
-	}
-
-	class MockLanguageClient implements LanguageClient {
-		MessageType expectedMessageType;
-		String message;
-		Boolean logWasReceived = false;
-
-		public MockLanguageClient(MessageType expectedMessageType, String message) {
-			this.expectedMessageType = expectedMessageType;
-			this.message = message;
-		}
-
-		public void telemetryEvent(Object object) {
-		}
-
-		@Override
-		public CompletableFuture<MessageActionItem> showMessageRequest(ShowMessageRequestParams requestParams) {
-			return null;
-		}
-
-		@Override
-		public void showMessage(MessageParams messageParams) {
-		}
-
-		@Override
-		public void publishDiagnostics(PublishDiagnosticsParams diagnostics) {
-		}
-
-		@Override
-		public void logMessage(MessageParams messageParams) {
-			assertEquals(expectedMessageType, messageParams.getType());
-			this.logWasReceived = true;
-		}
-
-		public Boolean wasLogReceived() {
-			return this.logWasReceived;
-		}
-
-	}
-
-	public void deleteLogFile() {
+	private void deleteLogFile() {
 		File f = new File(path);
 		if (f.exists()) {
 			f.delete();

